File: cipher.java
import java.util.Scanner;

    public class Cipher{
        public static void main(String[] args){
            System.out.println("Digite o texto a ser criptorafado: ");
            Scanner bucky = new Scanner(System.in);
            String criptoString = bucky.nextLine();
            criptoString = criptoString.toUpperCase();
            System.out.println("Deseja criptografar com Cifra de Cesar ou com Cifra de Vigenere?");
            System.out.println("Aperte 1 para Cesar e 2 para Vigenere: ");
            int op = bucky.nextInt();
            if(op == 1){
                System.out.println("Digite a chave a ser criptorafada: ");
                int criptoKey1 = bucky.nextInt();
                String x = Cesar(criptoString, criptoKey1);
                System.out.println(x); 
            }
            if(op == 2){
                System.out.println("Digite a chave a ser criptorafada: ");
                Scanner bucky2 = new Scanner(System.in);
                String criptoKey2 = bucky2.nextLine();
                criptoKey2 = criptoKey2.toUpperCase();
                String y = Vigenere(criptoString, criptoKey2);
                System.out.println(y); 
            }
            
        
        }

        public static String Cesar(String criptoString, int key){
            char[] letters = criptoString.toCharArray();
            for(int i = 0; i < criptoString.length(); i++){
                if(criptoString.codePointAt(i) >= 65 && criptoString.codePointAt(i) <= 90){
                    int number = ((criptoString.codePointAt(i)-65+key)%26) + 65;
                    letters[i] = (char)number;
                    /*char c = char(number);
                    criptoString.charAt(i) = c;*/
                }
            }
            return new String(letters);
        }
        
        public static String Vigenere(String criptoString, String key){
            int z = 0;
            char[] letters = criptoString.toCharArray();
            //char[] letters2 = key.toCharArray();
            for(int i = 0; i < criptoString.length(); i++){
                if(criptoString.codePointAt(i) >= 65 && criptoString.codePointAt(i) <= 90){
                    int number = ((criptoString.codePointAt(i)-65+key.codePointAt(z++%key.length())-65)%26)+65;
                    letters[i] = (char)number;
                }
            }
            
            return new String(letters);
        }
        
        
    }
    

File: cipher.py
def caesar(criptoString, key):
    list1 = list(criptoString)
    for i in range (len(criptoString)):
        if(str(list1[i]) < 'A' or str(list1[i]) > 'Z'):
            list1[i] = criptoString[i]
        else:
            a = int(ord(criptoString[i]))
            list1[i] = chr(((a - ord('A') + key)%26) + ord('A'))
    newString = ''.join(list1)
    print(newString)


def vigenere(criptoString, key):
    z = 0
    list2 = list(criptoString)
    for i in range(len(criptoString)):
        if(z == len(key)):
            z = 0
        if(str(list2[i]) < 'A' or str(list2[i]) > 'Z'):
            list2[i] = criptoString[i]
        else:
            a = int(ord(criptoString[i]))
            list2[i] = chr((a - ord('A') + ord(key[z]) - ord('A'))%26 + ord('A'));
            z = z+1

    newString = ''.join(list2)
    print(newString)


def main():
    op = int(input("Aperte 1 para Cifra de César e 2 para Cifra de Vigenere: "))
    criptoString = input()
    criptoString = criptoString.upper()
    if(op == 1):
        criptoKey1 = int(input())
        caesar(criptoString, criptoKey1)
    if(op == 2):
        criptoKey2 = input()
        criptoKey2 = criptoKey2.upper()
        vigenere(criptoString, criptoKey2)


main()